src/plugins/workspace/public/components/workspace_creator/workspace_form.tsx

```diff
@@ -324,11 +344,26 @@ export const WorkspaceForm = ({
         setFormErrors({ permissions: permissionErrors });
         return;
       }
+
+      const featureConfigChanged =
+        formData.features.length !== defaultFeatures.length ||
+        formData.features.some((feat) => !defaultFeatures.includes(feat));
+
+      if (!featureConfigChanged) {
+        // If feature config not changed, set workspace feature config to the original value.
+        // The reason why we do this is when a workspace feature is configured by wildcard,
+        // such as `['@management']` or `['*']`. The form value `formData.features` will be
+        // expanded to array of individual feature id, if the feature hasn't changed, we will
+        // set the feature config back to the original value so that category wildcard won't
+        // expanded to feature ids
+        formData.features = defaultValues?.features ?? [];
```

Comment:
Is that means the wildcard match will be removed after add some new feature ids?

Reply:
Yes, that makes sense to me as that's what the user selected

Reply:
OK. Just wonder if we can support wildcard for the feature category level. Which means if user select a feature category, we will store wildcard feature rule. If they remove some one, we will change to feature ids.

Reply:
Sounds good to me, we could have an improvement ticket for this.